File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 23/04/2026, 00:07:26
class Solution {
 public:
  bool isPalindrome(int x) {
    if (x < 0)
      return false;

    long reversed = 0;
    int y = x;

    while (y > 0) {
      reversed = reversed * 10 + y % 10;
      y /= 10;
    }

    return reversed == x;
  }
};

File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 23/04/2026, 00:07:30
class Solution {
    public boolean isPalindrome(int x) {
       int sum = 0, r;
	 int temp = x;    
     while(x>0)
	   {    
        r = x % 10;   
        sum = (sum*10)+r;    
        x = x/10;    
       }    
      if(temp==sum)    
        return(true);    
      else    
       return(false);   
     }  
    }
